src/emailVerification/index.tsx

```diff
@@ -0,0 +1,242 @@
+    let result: boolean = false;
+    //[api]이메일 중복 검사
+    return result;
+  };
+  const sendVerificationEmail = () => {
+    let result: boolean = true;
+    // [api] 인증 이메일 보내기
+    return result;
+  };
+  const onClickEmailBtn = async () => {
+    overTime && setOverTime(false);
+    const overSending: boolean = sendingEmailCount.current > 4;
+    // 1.가능한 이메일 인증 횟수를 충족한 경우
+    if (!overSending) {
+      //A. 백엔드에 이메일  중복 여부 확인
+      let duplicate: boolean = false; // 중복 이메일
+      if (isEmailDuplicate) {
+        const result = checkDuplicateEmail();
+        // result 값에 따라 duplicate 값 변경
+      }
+      // A-1 중복 메일인 경우
+      if (duplicate) {
+        setEmail((prev: InputDataType) => {
+          const newState: InputDataType = {
+            ...prev,
+            errorMsg: '이미 회원가입된 이메일이에요.',
+          };
+          return newState;
+        });
+      }
+      // A-2 유효한 메일
+      if (!duplicate) {
+        // a 서버에 이메일 인증 보내기
+        const result = sendVerificationEmail();
+        // 인증 번호 이메일 전송 성공
+        //  a-1 이메일 발송 성공
+        if (result) {
+          successSendingEmail.current = true;
+          // ㄱ. 타이머 작동. 모달 오픈
+          setOpenTimer(true);
+          if (nextBtnElDomRect !== null && nextBtnElDomRect !== undefined) {
+            // modal의 height, width에 따라 top,left 값 변경
+            setOpenToastModal(true);
+            setToastModalState({
+              contents: '인증 이메일이 발송됐어요.',
+              //  39:toastModal.height
+              top: `${nextBtnElDomRect.top - 39 - 16}px`,
+              left: `20vw`,
+            });
+            sendingEmailCount.current += 1;
+          }
+        }
+      }
+    }
+    //  2. 할 수 있는 이메일 인증 횟수를 초과한 경우
+    if (overSending) {
+      setOpenToastModal(false);
+      setTimeout(() => {
+        setOpenAlertModal(true);
+      }, 100);
+    }
+  };
+  const onChangeAuthNumber = (event: ChangeEvent<HTMLInputElement>) => {
+    const text = XSSCheck(event.target.value, undefined);
+    setAuthNumber(text);
+    if (checkAuthNumber) setCheckAuthNumber(false);
+  };
+  //서버에서 받은 데이터
+  const getAuthNumber = () => {
+    let result = '111111';
+    //[api]
+    return result;
+  };
+  const onClickAuthNumberBtn = async () => {
+    //백엔드에 이메인 인증 번호 확인
+    const result = getAuthNumber();
+    setCheckAuthNumber(true);
+    //data는  string type으로
+    if (authNumber !== undefined && result === authNumber) {
+      setPass(true);
+      setDisableBtn(false);
+      setOpenTimer(false);
+      verifiedEmail.current = email.value;
+    } else {
+      setPass(false);
+    }
+  };
+  useEffect(() => {
+    if (email.value !== '' && email.errorMsg === null) {
+      // 이미 인증이 완료 된 경우에 다음 버튼 클릭 가능
+      setPass(true);
+      setCheckAuthNumber(true);
+      verifiedEmail.current = email.value;
+      setDisableBtn(false);
+    }
+  }, []);
+  useEffect(() => {
+    const inputEle = document.querySelector('#input-email') as HTMLInputElement | null;
+    if (inputEle !== null) {
```

Comment:
다음과 같이 작성해줘도 좋을 것 같습니다.
```suggestion
    if (inputEle) {
```